server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3958,6 +3961,116 @@ public VlanVO doInTransaction(final TransactionStatus status) {
 
     }
 
+    @Override
+    public Vlan updateVlanAndPublicIpRange(UpdateVlanIpRangeCmd cmd) throws ConcurrentOperationException,
+            ResourceUnavailableException, ResourceAllocationException{
+
+        return  updateVlanAndPublicIpRange(cmd.getId(), cmd.getStartIp(),cmd.getEndIp(),
+                cmd.getGateway(),cmd.getNetmask());
+    }
+
+    @DB
+    @ActionEvent(eventType = EventTypes.EVENT_VLAN_IP_RANGE_UPDATE, eventDescription = "update vlan ip Range", async
+            = false)
+    public Vlan updateVlanAndPublicIpRange(final long id, String startIp,
+                                           String endIp,
+                                           String gateway,
+                                           String netmask) throws ConcurrentOperationException{
+
+        // verify parameters
+        VlanVO vlanRange = _vlanDao.findById(id);
+        if (vlanRange == null) {
+            throw new InvalidParameterValueException("Please specify a valid IP range id.");
+        }
+
+        if (gateway == null) {
+            gateway = vlanRange.getVlanGateway();
+        }
+
+        if (netmask == null) {
+            netmask = vlanRange.getVlanNetmask();
+        }
+
+        final String[] existingVlanIPRangeArray = vlanRange.getIpRange().split("-");
+        final String currentStartIP= existingVlanIPRangeArray[0];
+        final String currentEndIP = existingVlanIPRangeArray [1];
+
+        if(startIp == null){
+            startIp= currentStartIP;
+        }
+
+        if(endIp == null){
+            endIp= currentEndIP;
+        }
+
+        // Check if the VLAN has any allocated public IPs
+        final List<IPAddressVO> ips = _publicIpAddressDao.listByVlanId(id);
+        for (final IPAddressVO ip : ips) {
+            if (ip.getState()== IpAddress.State.Allocated){
```

Comment:
very good below, but.. does it make sense to look for the first used from the range for lower boundry and the highest for upper boundry, to facilitate admins? Now and exception is thrown for the first mismatch and some usefull information may be withheld from the user.

Reply:
Thought its okay, because it ensures the boundaries remain within range. Regarding the info that could be withheld? what did you have in mind?

Reply:
the fact that not just one IP address is a problem, but that there more?!

Reply:
Okay... so it should report all IPs that could potentially be a problem in one go? Both upper and lower that is? I'm I right?

Reply:
I don't think it is very user friendly to keep complaining on single Ip addresses one by one if there are say 20. Not saying anything of course is worse, so not sure if we should block this PR for it. I see you closed the issue, should I continue this discussion somewhere else?